### PR TITLE
Adding BicycleRouting helper

### PR DESCRIPTION
**Pull request by Ashish-uzumaki**
In the current version of bicycle routing helper, it can fix bicycle routes. it can fix all gaps and invalid ways as well as can fix branches(on some conditions) with roles forward and backward.

### CODE REVIEW

File: src/main/java/org/openstreetmap/josm/plugins/pt_assistant/PTAssistantPlugin.java [hunk 1]
```diff
@@ -192,7 +192,19 @@ public int order() {
                 return Arrays.asList(new SortPTRouteMembersAction(editorAccess));
             }
         };
+//        IRelationEditorActionGroup group3 = new IRelationEditorActionGroup() {
```
[the remark this answers is not in the document]
Comment (Ashish-uzumaki) [reply src/main/java/org/openstreetmap/josm/plugins/pt_assistant/PTAssistantPlugin.java]: group3 was just to test the class we don't need that. main RoutingAction is in group1

File: src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    editor.addWindowListener(new WindowEventHandler());
+  }
+  @Override
+  protected void updateEnabledState() {
+      final Relation curRel = relation;
+      setEnabled(
+          curRel != null && setEnable &&
+          (
+              (curRel.hasTag("route", "bus") && curRel.hasTag("public_transport:version", "2")) ||
+              (RouteUtils.isPTRoute(curRel) && !curRel.hasTag("route", "bus")) || (curRel.hasTag("route", "bicycle"))
```
[the remark this answers is not in the document]
Comment (Ashish-uzumaki) [reply src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java]: walking and horsing have different sets of traffic rules than a bicycle and public transport and they don't need forward and backward relations still, I will try  to make it work for horse and foot relation

File: src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  public void actionPerformed(ActionEvent e) {
+       if (relation.hasIncompleteMembers()) {
+           downloadIncompleteRelations();
+           new Notification(tr("Downloading incomplete relation members. Kindly wait till download gets over."))
+                   .setIcon(JOptionPane.INFORMATION_MESSAGE).setDuration(3600).show();
+       } else {
+    if(relation.hasTag("route","bicycle")) {
+    	System.out.println("I got bicycle access");
+        Bicycle bike = new Bicycle(editorAccess);
+        bike.init();
```
[the remark this answers is not in the document]
Comment (Ashish-uzumaki) [reply src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java]: Okay I will try to make BicycleRoutingHelper as a child of Mendrelationaction

File: test/unit/org/openstreetmap/josm/plugins/pt_assistant/AbstractTest.java [hunk 4]
```diff
@@ -62,6 +62,8 @@
 
  public static final String PATH_TO_SEGMENT_TEST = "test/data/segment-test.osm";
 
+ public static final String PATH_TO_BICYCLE_LOOP_BACK_TEST = "test/data/map4.osm";
```
[the remark this answers is not in the document]
Comment (Ashish-uzumaki) [reply test/unit/org/openstreetmap/josm/plugins/pt_assistant/AbstractTest.java]: I have added that file in the data folder now

File: src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ashish-uzumaki) [reply src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java]: I made the parent-child relation of bicycle and mend relation please check it again and let me know if you find anything wrong

File: src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (francois2metz) [reply src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java]: ok, that's fine.

File: src/main/java/org/openstreetmap/josm/plugins/pt_assistant/validation/BicycleFootRouteValidatorTest.java [hunk 5]
```diff
@@ -24,14 +24,15 @@
 public class BicycleFootRouteValidatorTest extends Test {
 
     public static final int ERROR_CODE_CONTINUITY = 3701;
-
+    // List<RelationMember> members;
     public BicycleFootRouteValidatorTest() {
         super(tr("Bicycle and foot routes Tests"));
+        // members.removeIf(m -> !m.isWay());
```
Comment (francois2metz) [inline src/main/java/org/openstreetmap/josm/plugins/pt_assistant/validation/BicycleFootRouteValidatorTest.java]: There is stil some commented code here.

File: src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+import org.openstreetmap.josm.tools.ImageProvider;
+import org.openstreetmap.josm.tools.Logging;
+import org.openstreetmap.josm.tools.Utils;
+
+public class RoutingAction extends AbstractRelationEditorAction{
+	static Relation relation = null;
+	MemberTableModel memberTableModel = null;
+	GenericRelationEditor editor = null;
+	List<RelationMember> members = null;
+	boolean setEnable = true;
```
[the remark this answers is not in the document]
Comment (francois2metz) [reply src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java]: The variable is still here but it's always `true`. What's the point of it?

**Comment by francois2metz** [comment]
Also, your branch has not been updated against upstream for a while and there is some conflicts. Could you merge or rebase master?

**Comment by don-vip** [comment]
The build [fails](https://josm.openstreetmap.de/jenkins/job/JOSM-Plugins/jdk=JDK8/1637/console):

```
init:
    [mkdir] Created dir: /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/build

pre-compile:

compile:
     [echo] compiling sources for /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/dist/pt_assistant.jar ...
    [javac] Compiling 64 source files to /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/build
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/ExtractPlatformNodeAction.java:105: warning: [deprecation] RemoveNodesCommand(Way,List<Node>) in RemoveNodesCommand has been deprecated
    [javac]                     UndoRedoHandler.getInstance().add(new RemoveNodesCommand(w, Collections.singletonList(nd)));
    [javac]                                                       ^
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTRouteSegment.java:37: warning: [UnusedVariable] Field 'relations' is never read.
    [javac]     private List<Relation> relations;
    [javac]                            ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTRouteSegment.java:34: warning: [UnusedVariable] Field 'AllStops' is never read.
    [javac]     private List<PTStop> AllStops;
    [javac]                          ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTStop.java:229: warning: [UnusedVariable] Local variable 'nodeName' is never read.
    [javac]             String nodeName = currentNode.get("name");
    [javac]                    ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line or 'currentNode.get("name");'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTStop.java:64: warning: [UnusedVariable] Field 'operator' is never read.
    [javac]     private String operator = "";
    [javac]                    ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTStop.java:61: warning: [UnusedVariable] Field 'routeRef' is never read.
    [javac]     private String routeRef = "";
    [javac]                    ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTStop.java:94: warning: [UnusedVariable] Local variable 'map' is never read.
    [javac]         MapFrame map = MainApplication.getMap();
    [javac]                  ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line or 'MainApplication.getMap();'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTStop.java:67: warning: [UnusedVariable] Field 'network' is never read.
    [javac]     private String network = "";
    [javac]                    ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTStop.java:55: warning: [UnusedVariable] Field 'ref' is never read.
    [javac]     private String ref = "";
    [javac]                    ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTStop.java:58: warning: [UnusedVariable] Field 'localRef' is never read.
    [javac]     private String localRef = "";
    [javac]                    ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTStop.java:493: warning: [UnusedVariable] Local variable 'flag2' is never read.
    [javac]             int flag1 = 0, flag2 = 0;
    [javac]                            ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTStop.java:227: warning: [UnusedVariable] Local variable 'platName' is never read.
    [javac]         String platName = platform.get("name");
    [javac]                ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line or 'platform.get("name");'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTWay.java:173: warning: [UnusedVariable] Local variable 'nodeName' is never read.
    [javac]             String nodeName = currentNode.get("name");
    [javac]                    ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line or 'currentNode.get("name");'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:94: warning: [UnusedMethod] Private method 'downloadIncompleteRelations' is never used.
    [javac]     private void downloadIncompleteRelations() {
    [javac]                  ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedMethod)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:823: warning: [UnusedVariable] Local variable 'num' is never read.
    [javac]             int num = WayUtils.findNumberOfCommonFirstLastNodes(lastBackWay, lastForWay);
    [javac]                 ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line or 'WayUtils.findNumberOfCommonFirstLastNodes(lastBackWay, lastForWay);'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:874: warning: [UnusedVariable] Local variable 'minWay' is never read.
    [javac]         Way minWay = w;
    [javac]             ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:956: warning: [UnusedVariable] Local variable 'w' is never read.
    [javac]                     Way w = super.members.get(i).getWay();
    [javac]                         ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line or 'super.members.get(i).getWay();'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:60: warning: [UnusedVariable] Field 'DEFAULT_DOWNLOAD_PARAMS' is never read.
    [javac]     private static final DownloadParams DEFAULT_DOWNLOAD_PARAMS = new DownloadParams();
    [javac]                                         ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line or 'static { new DownloadParams(); }'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:861: warning: [UnusedVariable] Local variable 'node2' is never read.
    [javac]         Node node2;
    [javac]              ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:824: warning: [UnusedVariable] Local variable 'nod' is never read.
    [javac]             Node nod;
    [javac]                  ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:819: warning: [UnusedVariable] Local variable 'flag2' is never read.
    [javac]         int flag = 0, flag2 = 0;
    [javac]                       ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:451: warning: [UnusedVariable] Local variable 'flag' is never read.
    [javac]         int flag = 0;
    [javac]             ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:859: warning: [UnusedVariable] Local variable 'roles' is never read.
    [javac]         String[] roles = new String[20];
    [javac]                  ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:910: warning: [UnusedVariable] Local variable 'flag' is never read.
    [javac]         int flag = 0;
    [javac]             ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:69: warning: [HidingField] Hiding fields of superclasses may cause confusion and errors. This field is hiding a field of the same name in superclass: MendRelationAction
    [javac]     boolean setEnable = true;
    [javac]             ^
    [javac]     (see https://errorprone.info/bugpattern/HidingField)
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/BicycleMendRelation.java:277: warning: [UnnecessaryParentheses] Unnecessary use of grouping parentheses
    [javac]         if ((way.hasTag("oneway:bicycle", acceptedTags)) && way.lastNode().equals(node)
    [javac]             ^
    [javac]     (see https://errorprone.info/bugpattern/UnnecessaryParentheses)
    [javac]   Did you mean 'if (way.hasTag("oneway:bicycle", acceptedTags) && way.lastNode().equals(node)'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/MendRelationAction.java:1640: warning: [UnusedVariable] Local variable 'wayToKeep' is never read.
    [javac]         Way wayToKeep = null;
    [javac]             ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/MendRelationAction.java:1622: warning: [UnusedVariable] Local variable 'w2' is never read.
    [javac]         Way w2 = null;
    [javac]             ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/MendRelationAction.java:1610: warning: [ModifiedButNotUsed] A collection or proto builder was created, but its values were never accessed.
    [javac]             List<Node> n = new ArrayList<>();
    [javac]                            ^
    [javac]     (see https://errorprone.info/bugpattern/ModifiedButNotUsed)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/DoubleSplitAction.java:164: warning: [UnusedVariable] Parameter 'previousAffectedWay' is never read.
    [javac]     private boolean firstNodeIsConnectionNode(Node node, Way affected, Way previousAffectedWay) {
    [javac]                                                                            ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean 'private boolean firstNodeIsConnectionNode(Node node, Way affected) {'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/DoubleSplitAction.java:175: warning: [UnusedVariable] Parameter 'node' is never read.
    [javac]     private boolean secondNodeIsConnectionNode(Node node, Way previousAffectedWay) {
    [javac]                                                     ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean 'private boolean secondNodeIsConnectionNode(Way previousAffectedWay) {'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/DoubleSplitAction.java:503: warning: [UnusedVariable] Parameter 'type' is never read.
    [javac]             JComboBox<String> values, int type) {
    [javac]                                           ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean 'addTags(affectedKeysList, Arrays.asList(selectedWay), keys, values);'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/DoubleSplitAction.java:471: warning: [UnusedVariable] Parameter 'commandList' is never read.
    [javac]     private void addKeysWhenStartEndPoint(Way affected, List<Command> commandList, JComboBox<String> keys, JComboBox<String> values) {
    [javac]                                                                       ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean 'private void addKeysWhenStartEndPoint(Way affected, JComboBox<String> keys, JComboBox<String> values) {'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/DoubleSplitAction.java:149: warning: [ModifiedButNotUsed] A collection or proto builder was created, but its values were never accessed.
    [javac]                         List<TagMap> affectedKeysList = new ArrayList<>();
    [javac]                                                         ^
    [javac]     (see https://errorprone.info/bugpattern/ModifiedButNotUsed)
    [javac]   Did you mean to remove this line or to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/EdgeSelectionAction.java:171: warning: [UnusedVariable] Parameter 'initial' is never read.
    [javac]     private String getModeOfTravel(Way initial) {
    [javac]                                        ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean 'private String getModeOfTravel() {'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java:53: warning: [UnnecessaryParentheses] Unnecessary use of grouping parentheses
    [javac]               (RouteUtils.isPTRoute(curRel) && !curRel.hasTag("route", "bus")) || (curRel.hasTag("route", "bicycle"))
    [javac]                                                                                   ^
    [javac]     (see https://errorprone.info/bugpattern/UnnecessaryParentheses)
    [javac]   Did you mean '(RouteUtils.isPTRoute(curRel) && !curRel.hasTag("route", "bus")) || curRel.hasTag("route", "bicycle")'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/RoutingAction.java:97: warning: [ClassCanBeStatic] Inner class is non-static but does not reference enclosing class
    [javac]   class WindowEventHandler extends WindowAdapter {
    [javac]   ^
    [javac]     (see https://errorprone.info/bugpattern/ClassCanBeStatic)
    [javac]   Did you mean 'static class WindowEventHandler extends WindowAdapter {'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/SortPTRouteMembersAction.java:712: warning: [UnusedVariable] Parameter 'ptstop' is never read.
    [javac]     private static void removeWrongSideStops(List<PTStop> ptstop, List<RelationMember> wayMembers) {
    [javac]                                                           ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean 'removeWrongSideStops(wayMembers);'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/SortPTRouteMembersAction.java:383: warning: [UnusedVariable] Local variable 'curr1' is never read.
    [javac]         Way curr1 = null;
    [javac]             ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/Edge.java:18: error: [SelfAssignment] Variable assigned to itself
    [javac]         color = color;
    [javac]               ^
    [javac]     (see https://errorprone.info/bugpattern/SelfAssignment)
    [javac]   Did you mean 'this.color = color;'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTRouteDataManager.java:183: warning: [UnusedMethod] Private method 'calculateDistanceSq' is never used.
    [javac]     private double calculateDistanceSq(RelationMember member1, LatLon coord2) {
    [javac]                    ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedMethod)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTRouteDataManager.java:44: warning: [UnusedVariable] Field 'routeNameFormat' is never read.
    [javac]     private String routeNameFormat = "[operator] [ref] [from] - [via] - [to]";
    [javac]                    ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTRouteDataManager.java:55: warning: [UnusedVariable] Field 'RAINBOW_COLOR_PALETTE' is never read.
    [javac]     private static final Color[] RAINBOW_COLOR_PALETTE = { new Color(0, 255, 0, 150), //GREEN 0
    [javac]                                  ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/utils/StopToWay.java:184: warning: [UnusedMethod] Private method 'calculateMinDistanceToSegment' is never used.
    [javac]     private double calculateMinDistanceToSegment(LatLon node, Way way) {
    [javac]                    ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedMethod)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/utils/StopToWay.java:394: warning: [UnusedVariable] Local variable 'w' is never read.
    [javac]         Way w = null;
    [javac]             ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/utils/StopToWay.java:374: warning: [UnusedVariable] Local variable 'indx' is never read.
    [javac]         int idx = 0, indx = 0;
    [javac]                      ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/utils/StopToWay.java:396: warning: [UnusedVariable] Local variable 'indx' is never read.
    [javac]         int idx = 0, indx = 0;
    [javac]                      ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/utils/StopToWay.java:372: warning: [UnusedVariable] Local variable 'w' is never read.
    [javac]         Way w = null;
    [javac]             ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/utils/StopToWayAssigner.java:292: warning: [UnusedVariable] Local variable 'minWaySegment' is never read.
    [javac]         Pair<Node, Node> minWaySegment;
    [javac]                          ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean to remove this line?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/utils/WayUtils.java:107: warning: [UnnecessaryParentheses] Unnecessary use of grouping parentheses
    [javac]             || (
    [javac]                ^
    [javac]     (see https://errorprone.info/bugpattern/UnnecessaryParentheses)
    [javac]   Did you mean '|| way.hasTag("highway", "pedestrian")'?
    [javac] /var/lib/jenkins/jobs/JOSM-Plugins/workspace/jdk/JDK8/pt_assistant/src/main/java/org/openstreetmap/josm/plugins/pt_assistant/validation/PTAssistantValidatorTest.java:309: warning: [UnusedVariable] Parameter 'id' is never read.
    [javac]     private int showProceedDialog(long id, int numberOfDirectionErrors, int numberOfRoadTypeErrors) {
    [javac]                                        ^
    [javac]     (see https://errorprone.info/bugpattern/UnusedVariable)
    [javac]   Did you mean 'userInput[0] = showProceedDialog(directionErrorParameter, roadTypeErrorParameter);'?
    [javac] 1 error
    [javac] 50 warnings

BUILD FAILED
```

**Pull request merged**